web/src/beta/components/fields/DateTimeField/EditPanel/hooks.ts

```diff
@@ -0,0 +1,77 @@
+import moment from "moment-timezone";
+import { useCallback, useState } from "react";
+
+type Props = {
+  onChange?: (value?: string | undefined) => void;
+};
+
+type TimezoneInfo = {
+  timezone: string;
+  offset: string;
+};
+
+export default ({ onChange }: Props) => {
+  const [time, setTime] = useState<string>("");
+  const [date, setDate] = useState<string>("");
+  const [timezones] = useState(moment.tz.names());
+  const [selectedTimezone, setSelectedTimezone] = useState("+0:00");
+
+  const handleTimeChange = useCallback((newValue: string | undefined) => {
+    if (newValue === undefined) return;
+    setTime(newValue);
+  }, []);
+
+  const handleDateChange = useCallback((newValue: string | undefined) => {
+    if (newValue === undefined) return;
+    setDate(newValue);
+  }, []);
+
+  const getUniqueTimezones = useCallback((timezones: string[]): TimezoneInfo[] => {
+    const uniqueTimezones: TimezoneInfo[] = [];
+    const seenOffsets = new Set<string>();
+    timezones.forEach(timezone => {
+      const offset = moment.tz(timezone).utcOffset() / 60;
+
+      if (Number.isInteger(offset)) {
+        const offsetString = offset >= 0 ? `+${offset}` : `${offset}`;
+        const offsetTimezone = `${offsetString}:00`;
+
+        if (!seenOffsets.has(offsetTimezone)) {
+          uniqueTimezones.push({ timezone, offset: offsetTimezone });
+          seenOffsets.add(offsetTimezone);
+        }
+      }
+    });
+
+    uniqueTimezones.sort((a, b) => {
+      const offsetA = parseInt(a.offset);
+      const offsetB = parseInt(b.offset);
+      return offsetA - offsetB;
+    });
+
+    return uniqueTimezones;
+  }, []);
+
+  const zones: string[] = moment.tz.names();
+  const offsetFromUTC: TimezoneInfo[] = getUniqueTimezones(zones);
```

Comment:
If I am wrong please let me know, but I feel like this will be relatively static, so can you pull these out of the hook? If moment must be used inside a hook or a react component, maybe its fine here. But it seems more like a util that we can import and use once in a useMemo and then it shouldn't re-render at all and keeps the code simpler.

Whether or not you do the above, also please:
- put `const zones: string[] = moment.tz.names();` inside getUniqueTimezones because its not being used anywhere else
- make offsetFromUTC a useMemo